### ISSUE TITLE
PFP::control - Configure Vdd VRM temp monitoring

### ISSUE THREAD

**Issue by msbarth** [ibm-openbmc/dev#2753]
From discussions on handling the Vdd VRM temp within the fan control algo, there are a few different choices in how the event(s) will be configured within fan control. Here's a snippet showing the options:
```
Issue overview:

The BMC VRM Monitoring Service polls the VRMs for telemetry information (temp, current, voltage) every X seconds.
Users may want to collect this same information via the CFT.  These commands per the POR go directly to the VRM and are not coordinated with the VRM service.
Users may want to write to the VRMs for voltage biasing via the CFT.  These commands per the POR go directly to the VRM and are not coordinated with the VRM service.
Because these operations are not atomic, bad things can happen to the VRMs if both the CFT and the VRM monitoring service are talking to the VRM.  Bad data and potentially bad voltage writes or worse.

The resolution plan:

1.  For data reads, the CFT will read cached values read by the VRM service.   The commands will not go directly to the VRM.  This avoids the contention.
2.  For data writes, the CFT will turn off the VRM monitoring service, write to the VRM, then turn the VRM monitoring back on.

Issue: When the VRM monitoring service has the monitoring turned off, the VDD VRM temperatures, used for fan control, will not be available.
Resolution:  This will be a short amount of time and the direction is that we will just ignore this window and the potential thermal issue.
Stretch Goal:  The BMC could use the VDD temperature from the OCC (as the FSP does).  This would mean the VDD temp would always be available.  This is the best solution as then the fan control and DVFS algorithm are operating on the same data.
Potential improvement :  The fan control could be configured to see if the VRM monitoring service is turned off and set the fans to high speed if it is.  This seems like a good safety net and is very little work.

Other requests and potential improvements:

1.  The request is for the VRM Monitoring Service to have a tunable sensor collection rate.  Shawn thinks this work is minimized if he just makes it data driven in the JSON file.  I agree that is the right design anyway.  Then they can just load a new JSON file to tweak this in characterization.
2.  There was a request for a "reset" of the mins and maxes.  I agree this is a good design as well, but the resource hit here may be too high.
```

**Comment by msbarth** [comment]
VDD object paths:
```
/xyz/openbmc_project/sensors/temperature/vdd_p0_dcm0_rail_temperature
/xyz/openbmc_project/sensors/temperature/vdd_p0_dcm1_rail_temperature
/xyz/openbmc_project/sensors/temperature/vdd_p1_dcm0_rail_temperature
/xyz/openbmc_project/sensors/temperature/vdd_p1_dcm1_rail_temperature
```

**Comment by msbarth** [comment]
Work is being done to have the VDD put on dbus from the OCC, which is the preferred way to retrieve that temp.

Completing this issue will be included once all the available OCC temps are available on dbus and #1982 is complete since this will allow for fan speed decreases.

**Comment by msbarth** [comment]
Resolved by:
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49396
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49397
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49399
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49400
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49401
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49402
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49446
- https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-fan-presence/+/49447
